app/services/subform-detail-service.js

```diff
@@ -0,0 +1,41 @@
+import Ember from "ember";
+import AjaxPromise from "./../utils/ajax-promise";
+import _ from "lodash";
+const { getOwner } = Ember;
+
+export default Ember.Service.extend({
```

Comment:
Good move creating a service for everything details related.

Note that there is a handy base API service you can use : 

```javascript
export default ApiBaseService.extend({
   ...

   updateRequest(paramsObj, previousValue) {
       this.PUT({ data })
    }
})
```

as well as an `async` mixin for everything related to the loading spinner : )